{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from sklearn.preprocessing import StandardScaler \n",
    "\n",
    "# Import dataset\n",
    "X_train = pd.read_csv(\"C:/Users/steve/Desktop/Notebooks/Thesis-Project/Datasets/processed/X_train_processed.csv\")\n",
    "X_test = pd.read_csv(\"C:/Users/steve/Desktop/Notebooks/Thesis-Project/Datasets/processed/X_test_processed.csv\")\n",
    "y_train = pd.read_csv(\"C:/Users/steve/Desktop/Notebooks/Thesis-Project/Datasets/processed/y_train_processed.csv\")\n",
    "y_test = pd.read_csv(\"C:/Users/steve/Desktop/Notebooks/Thesis-Project/Datasets/processed/y_test_processed.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "y_train = np.ravel(y_train)\n",
    "y_test = np.ravel(y_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0, 3], dtype=int64)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Group one\n",
    "group_one = y_train[y_train[\"Diagnosis\"] == 0]\n",
    "group_two = y_train[y_train[\"Diagnosis\"] == 3]\n",
    "\n",
    "combined_groups_one = pd.concat([group_one, group_two], ignore_index = True)\n",
    "combined_groups_one[\"Diagnosis\"].unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Is our target imbalanced for group one?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Diagnosis\n",
       "0    58.40708\n",
       "3    41.59292\n",
       "Name: proportion, dtype: float64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check percentage of each class\n",
    "combined_groups_one[\"Diagnosis\"].value_counts(normalize=True) * 100"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Not as much!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((425, 16), (183, 16))"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X_train.shape, X_test.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((425, 16), (183, 16))"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sc = StandardScaler()\n",
    "\n",
    "# let's transform the data \n",
    "X_train_scaled = sc.fit_transform(X_train)\n",
    "X_test_scaled = sc.transform(X_test)\n",
    "\n",
    "X_train_scaled.shape, X_test_scaled.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.svm import SVC\n",
    "from sklearn.tree import DecisionTreeClassifier\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "lg = LogisticRegression(multi_class = \"auto\", solver = \"lbfgs\", max_iter = 1000, random_state = 42)\n",
    "\n",
    "svm = SVC(kernel ='rbf', decision_function_shape ='ovr', probability = True, random_state = 42)\n",
    "\n",
    "dt = decision_tree_model = DecisionTreeClassifier(criterion ='gini', max_depth = 5, min_samples_split = 10,\n",
    "                                                  min_samples_leaf = 5, max_features = 'sqrt', random_state = 42)\n",
    "\n",
    "rf = RandomForestClassifier(n_estimators = 100, criterion = 'gini', max_depth = 5, min_samples_split = 10, \n",
    "                            min_samples_leaf = 5, max_features = 'sqrt', bootstrap = True, random_state = 42)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.metrics import make_scorer, accuracy_score, precision_score, recall_score, f1_score, roc_auc_score\n",
    "from sklearn.model_selection import KFold, cross_validate\n",
    "\n",
    "kf = KFold(n_splits = 5, shuffle = True, random_state = 42)\n",
    "\n",
    "# Define metrics to evaluate\n",
    "scoring_metrics = {\n",
    "    'accuracy': make_scorer(accuracy_score),\n",
    "    'precision': make_scorer(precision_score, average = 'weighted', zero_division=0),\n",
    "    'recall': make_scorer(recall_score, average = 'weighted', zero_division=0),\n",
    "    'f1': make_scorer(f1_score, average = 'weighted', zero_division=0),\n",
    "    'roc_auc': make_scorer(roc_auc_score)\n",
    "}\n",
    "\n",
    "models  = {\"Logistic Regression\": lg, \n",
    "           \"Support Vector Machine\": svm, \n",
    "           \"Decision Tree\": dt, \n",
    "           \"Random Forest\": rf\n",
    "}\n",
    "\n",
    "model_data_mapping = {\n",
    "    'Logistic Regression': X_train_scaled,\n",
    "    'Support Vector Machine': X_train_scaled,\n",
    "    'Decision Tree': X_train,\n",
    "    'Random Forest': X_train\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().\n",
      "  y = column_or_1d(y, warn=True)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().\n",
      "  y = column_or_1d(y, warn=True)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().\n",
      "  y = column_or_1d(y, warn=True)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().\n",
      "  y = column_or_1d(y, warn=True)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().\n",
      "  y = column_or_1d(y, warn=True)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().\n",
      "  y = column_or_1d(y, warn=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "------------------------------------------------------\n",
      "Logistic Regression\n",
      "Mean train accuracy: 0.6941176470588235 ± 0.01146681687624584\n",
      "Mean test accuracy: 0.64 ± 0.03121529214452141\n",
      "Mean train precision: 0.6871284262244898 ± 0.012349662904269539\n",
      "Mean test precision: 0.6531650575161011 ± 0.018219520864726575\n",
      "Mean train recall: 0.6941176470588235 ± 0.01146681687624584\n",
      "Mean test recall: 0.64 ± 0.03121529214452141\n",
      "Mean train f1: 0.6875366459443062 ± 0.011608383632557891\n",
      "Mean test f1: 0.6393745655926393 ± 0.027455935848831044\n",
      "Mean train roc_auc: nan ± nan\n",
      "Mean test roc_auc: nan ± nan\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().\n",
      "  y = column_or_1d(y, warn=True)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().\n",
      "  y = column_or_1d(y, warn=True)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().\n",
      "  y = column_or_1d(y, warn=True)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().\n",
      "  y = column_or_1d(y, warn=True)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "------------------------------------------------------\n",
      "Support Vector Machine\n",
      "Mean train accuracy: 0.7752941176470587 ± 0.009593827311941234\n",
      "Mean test accuracy: 0.583529411764706 ± 0.038375309247764895\n",
      "Mean train precision: 0.7749055251185821 ± 0.010843378163222764\n",
      "Mean test precision: 0.5800207410258946 ± 0.031212381300747563\n",
      "Mean train recall: 0.7752941176470587 ± 0.009593827311941234\n",
      "Mean test recall: 0.583529411764706 ± 0.038375309247764895\n",
      "Mean train f1: 0.766803545357318 ± 0.009690197203232474\n",
      "Mean test f1: 0.5691442896726835 ± 0.03606513035114108\n",
      "Mean train roc_auc: nan ± nan\n",
      "Mean test roc_auc: nan ± nan\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "------------------------------------------------------\n",
      "Decision Tree\n",
      "Mean train accuracy: 0.7005882352941176 ± 0.0158277929965573\n",
      "Mean test accuracy: 0.4988235294117647 ± 0.05292810296095055\n",
      "Mean train precision: 0.6959359884166603 ± 0.017541150550336504\n",
      "Mean test precision: 0.5029860465849751 ± 0.07862160642862086\n",
      "Mean train recall: 0.7005882352941176 ± 0.0158277929965573\n",
      "Mean test recall: 0.4988235294117647 ± 0.05292810296095055\n",
      "Mean train f1: 0.6811766023429656 ± 0.01960337050884977\n",
      "Mean test f1: 0.4760410026044801 ± 0.05625859696676388\n",
      "Mean train roc_auc: nan ± nan\n",
      "Mean test roc_auc: nan ± nan\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\base.py:1474: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().\n",
      "  return fit_method(estimator, *args, **kwargs)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\base.py:1474: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().\n",
      "  return fit_method(estimator, *args, **kwargs)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\base.py:1474: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().\n",
      "  return fit_method(estimator, *args, **kwargs)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\base.py:1474: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().\n",
      "  return fit_method(estimator, *args, **kwargs)\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\base.py:1474: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().\n",
      "  return fit_method(estimator, *args, **kwargs)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "------------------------------------------------------\n",
      "Random Forest\n",
      "Mean train accuracy: 0.8070588235294117 ± 0.01551871289208578\n",
      "Mean test accuracy: 0.6070588235294119 ± 0.03614656822526404\n",
      "Mean train precision: 0.8224642737350333 ± 0.01632202609883332\n",
      "Mean test precision: 0.599062218206253 ± 0.030309259206173195\n",
      "Mean train recall: 0.8070588235294117 ± 0.01551871289208578\n",
      "Mean test recall: 0.6070588235294119 ± 0.03614656822526404\n",
      "Mean train f1: 0.795473180072406 ± 0.018022300017848804\n",
      "Mean test f1: 0.581026339235055 ± 0.02435766322697153\n",
      "Mean train roc_auc: nan ± nan\n",
      "Mean test roc_auc: nan ± nan\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n",
      "c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\model_selection\\_validation.py:1011: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 137, in __call__\n",
      "    score = scorer._score(\n",
      "            ^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_scorer.py\", line 350, in _score\n",
      "    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_param_validation.py\", line 213, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\steve\\anaconda3\\Lib\\site-packages\\sklearn\\metrics\\_ranking.py\", line 633, in roc_auc_score\n",
      "    raise ValueError(\"multi_class must be in ('ovo', 'ovr')\")\n",
      "ValueError: multi_class must be in ('ovo', 'ovr')\n",
      "\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "for model_name, model in models.items():\n",
    "\n",
    "    X_train_to_use = model_data_mapping[model_name]\n",
    "    \n",
    "    results = cross_validate(model, \n",
    "                             X_train_to_use, \n",
    "                             y_train, \n",
    "                             scoring = scoring_metrics,\n",
    "                             return_train_score = True,\n",
    "                             cv = kf)\n",
    "    print(\"------------------------------------------------------\")\n",
    "    print(model_name)\n",
    "    for metric in scoring_metrics.keys():\n",
    "            print(f'Mean train {metric}:', np.mean(results[f'train_{metric}']), '±', np.std(results[f'train_{metric}']))\n",
    "            print(f'Mean test {metric}:', np.mean(results[f'test_{metric}']), '±', np.std(results[f'test_{metric}']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
